woocommerce/woocommerce-ios
# [Just In Time Messages] GET JITM network request

Closes: #7908 


## Description


We will display a Just In Time Message, when available, at the top of the My Store tab.

When the tab is loaded, we will request the messages for this location. To do this, we will make a GET request from `path=jetpack/v4/jitm` with `query={"message_path": "woomobile:my_store:admin_notices"}`

This PR covers the networking layer for that request.

### PR length
Apologies that this is over 500 lines. I could split it into two if that would be helpful, but I haven't because I think it would be more difficult to review. Please let me know if you'd prefer me to do it though, it's no problem at all. I would split it into:
1. [The model and mapper and associated tests](https://github.com/woocommerce/woocommerce-ios/pull/7910/files/322ae1d7a894dbb6e0d9283abcc45ff71dcbcb13)
2. [The remote and associated tests](https://github.com/woocommerce/woocommerce-ios/pull/7910/files/322ae1d7a894dbb6e0d9283abcc45ff71dcbcb13..2a035788eaed26a177740b80b91f123e41a31d4b)

## Testing instructions


Not independently testable, since it's a currently-unused network request. Unit tests should pass.

To test it, add the following to a store's implementation of existing action, and trigger it, e.g. the top of `OrderStore.fetchFilteredOrders`, then open the Orders tab:

```
        Task {
            do {
                let result = try await jitmRemote.loadAllJustInTimeMessages(for: siteID, messagePath: .init(app: .wooMobile, screen: "my_store", hook: .adminNotices))
                switch result {
                case .success(let messages):
                    print("JITM: recieved \(messages)")
                case .failure(let error):
                    print("JITM: failure \(error)")
                }
            }
        }
```

N.B. see pdfdoF-1uc-p2#comment-2581 for details of setting up your store to be eligible for the test JITM

## Screenshots


<img width="754" alt="Debugger `print()` output showing a parsed JITM" src="https://user-images.githubusercontent.com/2472348/197544869-d7f04186-eebe-4639-89e4-847389172317.png">


---
